crates/oxc_linter/src/lint_runner.rs

```diff
@@ -0,0 +1,116 @@
+use std::{ffi::OsStr, path::PathBuf, sync::Arc};
+
+use oxc_diagnostics::DiagnosticSender;
+
+use crate::{ConfigStore, LintService, LintServiceOptions, Linter, TsGoLintState};
+
+/// Unified runner that orchestrates both regular (oxc) and type-aware (tsgolint) linting.
+pub struct LintRunner {
+    /// Regular oxc linter
+    regular_linter: Option<Linter>,
+    /// Type-aware tsgolint
+    type_aware_linter: Option<TsGoLintState>,
+}
+
+/// Builder for LintRunner
+pub struct LintRunnerBuilder {
+    regular_linter: Option<Linter>,
+    type_aware_enabled: bool,
+    config_store: ConfigStore,
+    cwd: PathBuf,
+    silent: bool,
+}
+
+impl LintRunnerBuilder {
+    pub fn new(cwd: PathBuf, config_store: ConfigStore) -> Self {
+        Self { regular_linter: None, type_aware_enabled: false, config_store, cwd, silent: false }
+    }
+
+    #[must_use]
+    pub fn with_linter(mut self, linter: Linter) -> Self {
+        self.regular_linter = Some(linter);
+        self
+    }
+
+    #[must_use]
+    pub fn with_type_aware(mut self, enabled: bool) -> Self {
+        self.type_aware_enabled = enabled;
+        self
+    }
+
+    #[must_use]
+    pub fn with_silent(mut self, silent: bool) -> Self {
+        self.silent = silent;
+        self
+    }
+
+    /// # Errors
+    /// Returns an error if the type-aware linter fails to initialize.
+    pub fn build(self) -> Result<LintRunner, String> {
+        let type_aware_linter = if self.type_aware_enabled {
+            match TsGoLintState::try_new(&self.cwd, self.config_store.clone()) {
+                Ok(state) => Some(state.with_silent(self.silent)),
+                Err(e) => return Err(e),
+            }
+        } else {
+            None
+        };
+
+        Ok(LintRunner {
+            regular_linter: self.regular_linter,
+            type_aware_linter,
+            // config_store: self.config_store,
+            // cwd: self.cwd,
+            // silent: self.silent,
+        })
+    }
+}
+
+impl LintRunner {
```

Comment:
Should the language server use this too? It looks like you had this in mind :)
The only difference should be the `DiagnosticSender`, the language server wants the `Message(WithPosition)` with the fix.

Reply:
yeah this is my intention, is the lang server will re-use `LintRuner`, we can possibly move config resolution into this as well and share it between the two